plasma_framework/contracts/src/transactions/GenericTransaction.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity 0.5.11;
+
+import "../utils/RLPReader.sol";
+
+/**
+ * @title GenericTransaction
+ * @notice GenericTransaction is a generic transaction format that makes few assumptions about the
+ * content of the transaction. At minimum a transaction must:
+ * - Be a list of 4 items: [txType, inputs, outputs, txData]
+ * - `txType` must be a uint not equal to zero
+ * - inputs must be a list
+ * - outputs must be a list
+ * - no assumptions are made about `txData`. Note that `txData` can be a list.
+ *
+ * It is expected that most transaction types will have similar outputs, so convenience methods for
+ * decoding outputs are provided. However, transactions types are free to extend this output format
+ * with extra data.
+ */
+library GenericTransaction {
+
+    uint8 constant private TX_NUM_ITEMS = 4;
+
+    using RLPReader for bytes;
+    using RLPReader for RLPReader.RLPItem;
+
+    struct Transaction {
+        uint256 txType;
+        RLPReader.RLPItem[] inputs;
+        RLPReader.RLPItem[] outputs;
+        RLPReader.RLPItem txData;
+    }
+
+    struct Output {
+        uint256 outputType;
+        bytes20 outputGuard;
+        address token;
+        uint256 amount;
+    }
+
+    /**
+    * @dev Decodes an RLP encoded transaction into the generic format.
+    */
+    function decode(bytes memory transaction) internal pure returns (Transaction memory) {
+        RLPReader.RLPItem[] memory rlpTx = transaction.toRlpItem().toList();
+        require(rlpTx.length == TX_NUM_ITEMS, "Invalid encoding of transaction");
+        uint256 txType = rlpTx[0].toUint();
+        require(txType > 0, "Transaction type must not be 0");
```

Comment:
Really minor issue: We are mixing `!=0` and `>0` when we check for fields in transactions. Could we stick to either the value must be greater than zero or the value can not be zero?